### DISCUSSION TITLE
Color coded glucose display values

### DISCUSSION THREAD

**Discussion by fantonsen** [j-kaltes/Juggluco#182] [Ideas]
Hi.
I just started to use Juggluco instead of Abbot Libre 3, and are very happy about the widget that makes my glucose value visible on the homepage of my phone without the need of opening the app.

However, the widget in use does not look as good as the preview of the widget. (At least on my phone)

This is the preview:
![Juggluco-Widget-Preview-small](https://github.com/user-attachments/assets/fe1b6c90-a9ce-475e-9997-659032db44da)

And here is the result on my phone:
![Juggluco-Widget](https://github.com/user-attachments/assets/d2db0263-11df-4832-b36c-4f3d31c6be2b)

The main difference is that the widget is transparent, and not always very readable without dark and even backgrounds on the phone.

My suggestion is to us a background-color as  shown in the preview, and even better is it to use different colors based on different good, high and low values. This would make it very visible when your values are a bit high or low. (before the alarms may go off)
The best and easiest would be to use the same (or similar) colors that are used on the statistics-page.

It would also be nice to have the same background color around the value in the app itself, like this:

Current version:
![Juggluco-App-Current](https://github.com/user-attachments/assets/41ffe7bd-70b5-40ca-988a-b4dc20789066)

Suggestion:
![Juggluco-App-Suggestion](https://github.com/user-attachments/assets/989c5d1d-a898-4066-b0dd-c3eb1c5f5b40)

(At least as an option)

**Comment by j-kaltes** [comment]
Why don't you use floating glucose (Left menu->Setting->Floating glucose) instead of the widget? You can give its background different colors.

**Comment by fantonsen** [comment]
I tried the floating glucose, but I still prefer the widget one.
With floating glucose, the value is visible on top of all other apps as well, and that is too disturbing for other apps. And really not what I need.
Having it visible only on the homepage of the phone is what I want, just like the widget does.

Also, none of them changes the colors automatically based on current glucose values as I suggested.
( Green for good values, red for low values, yellow/orange for high values )

Anyway, I have now changed my background picture of my phone to better suit the juggluco widget, and will use it as it is, regardless of this suggestion being implemented or not. :smiley:

**Reply by j-kaltes** [reply]
When there is no large empty space above another app to place the floating glucose, you can long press the value side of the floating glucose, thereby making it small. Touching it makes it large again.
